util/process_wrapper/system_posix.cc

```diff
@@ -174,12 +175,18 @@ int System::Exec(const System::StrType &executable,
     }
   }
 
-  int err, exit_status = -1;
+  int err, exit_status;
   do {
     err = waitpid(child_pid, &exit_status, 0);
   } while (err == -1 && errno == EINTR);
 
-  return exit_status;
+  if (WIFEXITED(exit_status)) {
```

Comment:
What is the fall-through case? Is it possible that we reach the end of the function without returning anything?

Reply:
+1, I'd add a fall through case. Otherwise LGTM, great work!